src/AMAMM.sol

```diff
@@ -67,12 +67,16 @@ contract AMAMM is IAmAmm {
         return poolEpochManager[id][epoch];
     }
 
-    function getManager(PoolId id, uint40 epoch) public view returns (Bid memory) {
-        //@dev getEpoch(id, epoch) shoudl always be smaller than lastupdated
-        if (_lastUpdatedEpoch[id] - _getEpoch(id, epoch) <= K(id)) {
-            return poolEpochManager[id][_lastUpdatedEpoch[id]];
+    function getLastManager(PoolId id, uint40 targetEpoch) public view returns (Bid memory) {
+        if (_lastUpdatedEpoch[id] > targetEpoch) {
+            if (_lastUpdatedEpoch[id] > K(id)) return poolEpochManager[id][_lastUpdatedEpoch[id] - K(id)];
```

Comment:
```suggestion
            if (_lastUpdatedEpoch[id] - targetEpoch <= K(id)) return poolEpochManager[id][_lastUpdatedEpoch[id]];
```
Not sure why we need to minus K as it should just return the `_lastUpdatedEpoch[id]`